pkg/components/ebpf/common/redis_detect_transform.go

```diff
@@ -190,11 +193,37 @@ func redisStatus(buf []byte) (request.DBError, int) {
 	return dbError, status
 }
 
-func TCPToRedisToSpan(trace *TCPRequestInfo, op, text string, status int, dbError request.DBError) request.Span {
+func getRedisDB(connInfo BpfConnectionInfoT, op, text string, dbCache *simplelru.LRU[BpfConnectionInfoT, int]) int {
+	if dbCache == nil {
+		return -1
+	}
+	db, found := dbCache.Get(connInfo)
+	if !found {
+		db = -1
+	}
+	switch strings.ToUpper(op) {
+	case "SELECT":
+		// get db number from text after first space
+		if text != "" {
+			parts := strings.Split(text, " ")
+			if len(parts) > 1 {
+				if dbNum, err := strconv.Atoi(parts[1]); err == nil {
```

Comment:
can we do some sanity checks on the database number? I know that by default the dbs are 16, not sure if they can be increased via config

Reply:
ouch :D since it's an int maybe just check is not < 0?

Reply:
yes will do 👌